prow/external-plugins/needs-rebase/plugin/plugin.go

```diff
@@ -133,17 +136,45 @@ func HandleAll(log *logrus.Entry, ghc githubClient, config *plugins.Configuratio
 		log.Warnf("No repos have been configured for the %s plugin", PluginName)
 		return nil
 	}
-	var buf bytes.Buffer
-	fmt.Fprint(&buf, "archived:false is:pr is:open")
+	// GitHub hard caps queries at 1k results, so do one query per org and one for
+	// all repos. Ref: https://github.community/t/graphql-github-api-how-to-get-more-than-1000-pull-requests/13838/11
+	var queries []string
 	for _, org := range orgs {
-		fmt.Fprintf(&buf, " org:\"%s\"", org)
+		// https://img.17qq.com/images/crqhcuueqhx.jpeg
+		if org == "kubernetes" {
```

Comment:
This is unfortunate for k8s/kubernetes, and hopefully won't happen in any other repo. When it does happen we might want to make this configurable like `needs-rebase: large-org: true`, but for now 🤷‍♂️

Reply:
Yeah, so obviously this is pretty clunky, but it would be more than just a bool to plug through, we'd need to tell which repo to exclude from the org query and how to shard the query for that repo